src/inmanta/server/services/environment_metrics_service.py

```diff
@@ -139,7 +139,10 @@ async def start(self) -> None:
         self.register_metric_collector(ResourceCountMetricsCollector())
         self.register_metric_collector(AgentCountMetricsCollector())
         self.register_metric_collector(CompileTimeMetricsCollector())
-        self.schedule(self.flush_metrics, COLLECTION_INTERVAL_IN_SEC, initial_delay=0, cancel_on_stop=True)
+        self.previous_timestamp = datetime.now()
```

Comment:
What's the rationale behind setting this here as well as in the `__init__` ?

Reply:
The one in `__init__` is actually not required. Only the one here is required. But I didn't want to make `self.previous_timestamp` optional, so I set it in both locations.

Reply:
Giving it a second though, this line is not really required here. The one in `__init__.py` should be sufficient. I will remove this one.